tests/rbr/MakeRegionsTest.java

```diff
@@ -0,0 +1,97 @@
+package rbr;
+
+import java.util.*;
+import org.junit.*;
+import util.*;
+
+public class MakeRegionsTest {
+
+    private Set<Character> inputPorts = new HashSet<>();
+    private Set<Character> outputPorts = new HashSet<>();
+    private Set<Vertex> destinations = new HashSet<>();
+    private RBR rbr;
+
+    @Before
+    public void setUp() {
+        Graph graph = RegularGraphBuilder.generateGraph(4,4);
+        rbr = new RBR(graph);
+        inputPorts.add('I');
+        inputPorts.add('N');
+        inputPorts.add('E');
+        outputPorts.add('N');
+        outputPorts.add('S');
+        outputPorts.add('W');
+        destinations.add(new Vertex("0.1"));
+        destinations.add(new Vertex("0.2"));
+        destinations.add(new Vertex("0.3"));
+        destinations.add(new Vertex("1.0"));
+        destinations.add(new Vertex("1.2"));
+        destinations.add(new Vertex("1.3"));
+        destinations.add(new Vertex("2.0"));
+        destinations.add(new Vertex("2.3"));
+        destinations.add(new Vertex("3.0"));
+        destinations.add(new Vertex("3.1"));
+        destinations.add(new Vertex("3.2"));
+        destinations.add(new Vertex("3.3"));
+    }
+
+    @Test
+    public void emptyDestinationsCreatesNoRegion() throws Exception {
+        List<Region> createdRegionsWithoutDestinations = rbr.makeRegions(new HashSet<>(), inputPorts, outputPorts);
+        Assert.assertTrue(createdRegionsWithoutDestinations.isEmpty());
+    }
+
+    @Test
+    public void createdRegionsHaveSameInputPortsOriginals() throws Exception {
```

Comment:
Make the tests' name more 'sentence like'. Something as "createdRegionsHaveOriginalInputPorts".